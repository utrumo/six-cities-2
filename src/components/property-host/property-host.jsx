import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import {ASSETS_PATCH} from '../../shared/const';

const PropertyHost = (props) => {
  const {name, isPro, url, description} = props;
  const classes = classNames(
      `property__avatar-wrapper`,
      {'property__avatar-wrapper--pro': isPro},
      `user__avatar-wrapper`
  );
  const src = `${ASSETS_PATCH}${url}`;

  return (
    <div className="property__host">
      <h2 className="property__host-title">Meet the host</h2>
      <div className="property__host-user user">
        <div className={classes}>
          <img
            className="property__avatar user__avatar"
            src={src}
            width="74"
            height="74"
            alt="Host avatar"
          />
        </div>
        <span className="property__user-name">{name}</span>
        {isPro && <span className="property__user-status">Pro</span>}
      </div>
      <div className="property__description">
        <p className="property__text">{description}</p>
      </div>
    </div>
  );
};

PropertyHost.propTypes = {
  name: PropTypes.string.isRequired,
  isPro: PropTypes.bool.isRequired,
  url: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired
};

export default PropertyHost;
