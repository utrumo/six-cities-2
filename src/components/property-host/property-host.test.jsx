import React from 'react';
import renderer from 'react-test-renderer';
import {PropertyHost} from './property-host.jsx';

it(`PropertyPrice should renders correctly`, () => {
  const tree = renderer
    .create(
        <PropertyHost name="Alex" isPro={true} url="pictures/img.jpg" description="Welcome!" />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
