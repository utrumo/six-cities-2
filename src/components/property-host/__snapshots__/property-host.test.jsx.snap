// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyPrice should renders correctly 1`] = `
<div
  className="property__host"
>
  <h2
    className="property__host-title"
  >
    Meet the host
  </h2>
  <div
    className="property__host-user user"
  >
    <div
      className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
    >
      <img
        alt="Host avatar"
        className="property__avatar user__avatar"
        height="74"
        src="/pictures/img.jpg"
        width="74"
      />
    </div>
    <span
      className="property__user-name"
    >
      Alex
    </span>
    <span
      className="property__user-status"
    >
      Pro
    </span>
  </div>
  <div
    className="property__description"
  >
    <p
      className="property__text"
    >
      Welcome!
    </p>
  </div>
</div>
`;
